modules/core/shared/src/main/scala/gem/Observation.scala

```diff
@@ -27,8 +26,13 @@ final case class Observation[+T, +S, +D](
 
 object Observation {
 
-  /** A fully specified observation. */
-  type Full = Observation[TargetEnvironment, StaticConfig, Step[DynamicConfig]]
+  /** A fully specified observation, with unknown instrument. */
+  type Full = Full.Aux[I] forSome { type I <: Instrument with Singleton }
```

Comment:
Yes, I think it achieves what we want.

If we try to change an unadorned `Observation.Full` we find that we can't. The resulting type is no longer an `Observation.Full`.

```scala
def edit(o: Observation.Full): Observation.Full =
  o.copy(staticConfig = StaticConfig.Gpi())

error: type mismatch;
 found   : gem.Observation[gem.TargetEnvironment.Aux[I(in type Full)],gem.config.StaticConfig.Gpi,gem.Step[gem.config.DynamicConfig.Aux[(some other)I(in type Full)]]]
    (which expands to)  gem.Observation[gem.TargetEnvironment{type +I = I(in type Full)},gem.config.StaticConfig.Gpi,gem.Step[gem.config.DynamicConfig{type +I = (some other)I(in type Full)}]]
 required: gem.Observation.Full
    (which expands to)  gem.Observation[gem.TargetEnvironment{type I = I},gem.config.StaticConfig{type I = I},gem.Step[gem.config.DynamicConfig{type I = I}]]( forSome { type I <: gem.enum.Instrument with Singleton })
```

This *does* work if we want to muck around with a narrow-typed observation.
```scala
type GpiObservation = Observation.Full.Aux[Instrument.Gpi.type]

def edit(o: GpiObservation): GpiObservation =
  o.copy(staticConfig = StaticConfig.Gpi()) // static config is the right type
```

This also works if we want to edit a generic observation.

```scala
def edit2[I <: Instrument with Singleton](
  o: Observation.Full.Aux[I],
  s: StaticConfig.Aux[I]
): Observation.Full.Aux[I] = 
  o.copy(staticConfig = s)
```

Reply:
Is a plain `Observation.Full` useful for anything?  Should it even exist?  Sorry to be so dense, it's just unfamiliar. 

`ObservationDao.select` returns an `Observation.Full`.  You can't know a priori what the instrument is so that makes sense.  But then how to you narrow it down to a specific instrument in order to work with it?  I suppose a match on the instrument type, somehow, must be involved but I doubt that's correct.

Reply:
Yeah that's a good point. I'll add a narrowing operation.